src/hotspot/share/opto/loopnode.cpp

```diff
@@ -802,14 +802,14 @@ bool PhaseIdealLoop::create_loop_nest(IdealLoopTree* loop, Node_List &old_new) {
   // The number of iterations for the integer count loop: guarantee no
   // overflow: max_jint - stride_con max. -1 so there's no need for a
   // loop limit check if the exit test is <= or >=.
-  int iters_limit = max_jint - ABS(stride_con) - 1;
+  int iters_limit = max_jint - uabs(stride_con) - 1;
```

Comment:
It appears iters_limit can become -1 or -2 here, depending on the value of stride_con.  See below for the problem.